upgrade.py

```diff
@@ -487,6 +487,34 @@ def buildRestoreList(self, src_base):
         return restore_list
 
     completeUpgradeArgs = ['mounts', 'installation-to-overwrite', 'primary-disk', 'backup-partnum', 'logs-partnum', 'net-admin-interface', 'net-admin-bridge', 'net-admin-configuration']
+
+    def convert_interface_rename_data(self, src_file_path, dest_file_path):
+        """
+        Convert data of interface-rename to the data consumed by networkd.
+        """
+        net_devs = netutil.net_devs_of_last_boot(src_file_path)
+        if net_devs is None:
+            logger.error('Failed to parse the interface-rename dynamic rules.')
+            return False
+        else:
+            pattern = r'eth(\d+)'
+            lines = []
+            for (name, mac) in net_devs:
+                match = re.search(pattern, name)
+                if match:
+                    position = match.group(1)
+                    line = f"{position}:mac=\"{mac}\"\n"
+                    logger.log(f"Converted interface-rename data: {line}")
+                    lines += line
```

Comment:
Does this actually work? `lines` is a list and `line` is a string and the result of adding them is unlikely to be what you want:

```
>>> x = []
>>> x += "hello"
>>> x
['h', 'e', 'l', 'l', 'o']
```

Reply:
OK. It seems this works (as I tested it as a whole upgrade) but not in expected way like this.
I can make it write line by line as you are suggesting in another comment.